workflows/index-generation/index-generation.wdl

```diff
@@ -767,94 +679,54 @@ task CompressNT {
     command <<< 
         set -euxo pipefail
 
-        mkdir ~{split_apart_taxid_dir_name}
-
-        if [ "~{logging_enabled}" ]; then
-            ncbi-compress fasta-compress-from-taxid-dir \
-                --input-fasta-dir ~{reads_by_taxid} \
-                --output-fasta nt_compressed.fa \
-                --k ~{k} \
-                --scaled ~{scaled} \
-                --similarity-threshold ~{similarity_threshold} \
-                --split-apart-taxid-dir-name ~{split_apart_taxid_dir_name} \
-                --enable-sequence-retention-logging \
-                --logging-contained-in-tree-fn nt_contained_in_tree.tsv \
-                --logging-contained-in-chunk-fn nt_contained_in_chunk.tsv
-        else
-            ncbi-compress fasta-compress-from-taxid-dir \
-                --input-fasta-dir ~{reads_by_taxid} \
-                --output-fasta nt_compressed.fa \
-                --k ~{k} \
-                --scaled ~{scaled} \
-                --similarity-threshold ~{similarity_threshold} \
-                --split-apart-taxid-dir-name ~{split_apart_taxid_dir_name}
-        fi
-    >>>
-
-    output {
-        File nt_compressed = "nt_compressed.fa"
-        File nt_contained_in_tree = "nt_contained_in_tree.tsv"
-        File nt_contained_in_chunk = "nt_contained_in_chunk.tsv"
-        Directory split_taxid_dir = split_apart_taxid_dir_name
-    }
-
-    runtime {
-        docker: docker_image_id
-        cpu: 64
-        memory: "488G"
-    }
-}
+        READS_BY_TAXID_PATH=reads_by_taxid
+        SPLIT_APART_TAXID_DIR_NAME=split_apart_taxid
 
-task CompressNR {
-    input {
-        Directory reads_by_taxid
-        String split_apart_taxid_dir_name = "split_taxid_dir_nr"
-        Int k
-        Int scaled
-        Float similarity_threshold
-        Boolean logging_enabled
-        String docker_image_id
-    }
-
-    command <<<
-        set -euxo pipefail
+        # It is critical that this split happens in the same step as the compression
+        # If the directory is a step output it will be uploaded, which takes an enormous amount of time
```

Comment:
interesting, i wonder if this was what was creating the mysterious docker exit code -1 that i was seeing (upload timeout instead of disk space usage that we were initially thinking).

Reply:
I'm curious to hear about how you figured this out because i wouldn't have thought of this as an issue